tls/s2n_x509_validator.c

```diff
@@ -513,21 +587,27 @@ static S2N_RESULT s2n_x509_validator_verify_cert_chain(struct s2n_x509_validator
                 S2N_ERR_INTERNAL_LIBCRYPTO_ERROR);
     }
 
-    uint64_t current_sys_time = 0;
-    RESULT_GUARD(s2n_config_wall_clock(conn->config, &current_sys_time));
-    if (sizeof(time_t) == 4) {
-        /* cast value to uint64_t to prevent overflow errors */
-        RESULT_ENSURE_LTE(current_sys_time, (uint64_t) MAX_32_TIMESTAMP_NANOS);
-    }
+    if (conn->config->disable_x509_validity_period_validation) {
+        RESULT_GUARD(s2n_x509_validator_disable_validity_period_validation(conn, validator));
+    } else {
+        uint64_t current_sys_time = 0;
+        RESULT_GUARD(s2n_config_wall_clock(conn->config, &current_sys_time));
+        if (sizeof(time_t) == 4) {
+            /* cast value to uint64_t to prevent overflow errors */
+            RESULT_ENSURE_LTE(current_sys_time, (uint64_t) MAX_32_TIMESTAMP_NANOS);
+        }
 
-    /* this wants seconds not nanoseconds */
-    time_t current_time = (time_t) (current_sys_time / ONE_SEC_IN_NANOS);
-    X509_STORE_CTX_set_time(validator->store_ctx, 0, current_time);
+        /* this wants seconds not nanoseconds */
+        time_t current_time = (time_t) (current_sys_time / ONE_SEC_IN_NANOS);
+        X509_STORE_CTX_set_time(validator->store_ctx, 0, current_time);
+    }
```

Comment:
I'd put the explanation from your PR description about X509_STORE_CTX_set_time overriding the flag in a comment here. Does it have the same effect on the callback?

Reply:
Added! This doesn't have the same effect on the callback, so we could call `X509_STORE_CTX_set_time()` in the case that the `NO_CHECK_TIME` flag isn't supported. However, I'm not sure there's a benefit in doing this since all time errors are ignored anyway, so it shouldn't matter what the time is set to.